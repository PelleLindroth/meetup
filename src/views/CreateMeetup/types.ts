import { User } from '../../db/users'
import { Meetup } from '../../db/meetups'

export type CreateMeetupProps = {
  user: User
  meetups: Meetup[]
  setMeetups: React.Dispatch<React.SetStateAction<Meetup[]>>
}
