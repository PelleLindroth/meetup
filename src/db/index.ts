import { meetups, Meetup } from './meetups'
import { users, User } from './users'

export const getAllMeetups = (): Meetup[] => {
  const storedMeetups: Meetup[] = JSON.parse(localStorage.getItem('meetups')!)
  let returnedMeetups: Meetup[] = []

  if (storedMeetups) {
    returnedMeetups.push(...storedMeetups)
  } else returnedMeetups.push(...meetups)

  returnedMeetups.sort((a, b) => {
    if (a.date.getTime() > b.date.getTime()) {
      return 1
    } else if (b.date.getTime() > a.date.getTime()) {
      return -1
    } else {
      return 0
    }
  })

  return returnedMeetups
}

export const getMockMeetups = () => {
  meetups.sort((a, b) => {
    if (a.date.getTime() > b.date.getTime()) {
      return 1
    } else if (b.date.getTime() > a.date.getTime()) {
      return -1
    } else {
      return 0
    }
  })

  return meetups
}

export const saveMeetupToLocalStorage = (meetup: Meetup) => {
  const storedMeetups: Meetup[] = JSON.parse(localStorage.getItem('meetups')!)

  if (storedMeetups) {
    storedMeetups.push(meetup)

    localStorage.setItem('meetups', JSON.stringify(storedMeetups))
  } else {
    localStorage.setItem('meetups', JSON.stringify([meetup]))
  }
}

export const getMeetupById = (id: string): Meetup | undefined => {
  return meetups.find((meetup) => meetup.id === id)
}

export const validateUser = (email: string, password: string): User | null => {
  const user = users.find((user) => user.email === email)

  if (user) {
    return user.password === password ? user : null
  } else {
    return null
  }
}

export const getUserById = (id: string): User | undefined => {
  return users.find((user) => user.id === id)
}

export const getMockUsers = () => {
  return users
}
