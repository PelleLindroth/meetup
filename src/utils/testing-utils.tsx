import { render } from '@testing-library/react'
import { ReactElement } from 'react'
import { MemoryRouter } from 'react-router-dom'
import { Routes, Route } from 'react-router'

export const renderWithRouter = (ui: ReactElement, options?: any) =>
  render(ui, { wrapper: MemoryRouter, ...options })

export const renderWithPath = (ui: ReactElement, entry: string, path: string) =>
  render(
    <MemoryRouter initialEntries={[entry]}>
      <Routes>
        <Route path={path} element={ui} />
      </Routes>
    </MemoryRouter>
  )
