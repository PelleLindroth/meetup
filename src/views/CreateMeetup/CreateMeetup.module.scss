.createMeetup {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 5rem;

  h1 {
    font-size: 1.4rem;
  }

  form {
    display: flex;
    flex-direction: column;
    width: 80%;

    .inputRow {
      display: flex;
      flex-direction: column;
      margin-bottom: 1rem;

      label {
        font-weight: 700;
        margin-bottom: 0.2rem;
      }

      input,
      textarea {
        border: none;
        font-family: inherit;
        font-size: 1rem;
        outline: none;
      }

      .textInput {
        padding: 0.5rem;
      }

      .titleInput {
        font-weight: 700;
      }

      textarea {
        height: 3rem;
        resize: none;
      }
    }
  }
}
