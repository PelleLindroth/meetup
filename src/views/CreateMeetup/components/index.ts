export { default as AddressSection } from './AddressSection'
export { default as CapacitySection } from './CapacitySection'
export { default as DetailsSection } from './DetailsSection'
export { default as KeywordSection } from './KeywordSection'
export { default as SubmitSection } from './SubmitSection'
