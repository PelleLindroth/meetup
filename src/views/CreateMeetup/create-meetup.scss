.react-datetime-picker {
  background-color: #fff;

  .react-datetime-picker__wrapper {
    border: none;
    border-radius: 0.5rem;
    height: 2rem;
  }
  .react-datetime-picker__inputGroup__input {
    outline: none;
  }
}
