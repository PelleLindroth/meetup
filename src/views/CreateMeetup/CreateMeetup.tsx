import React, { useContext, useState } from 'react'
import { uid } from 'uid'
import { useNavigate } from 'react-router'
import { DateContext } from '../../contexts/DateContext'
import { addMeetup } from '../../db'
import { Meetup } from '../../db/meetups'
import { CreateMeetupProps } from './types'
import DateTimePicker from 'react-datetime-picker'
import styles from './CreateMeetup.module.scss'
import './create-meetup.scss'

const CreateMeetup = (props: CreateMeetupProps) => {
  const { user, meetups, setMeetups } = props
  const navigate = useNavigate()
  const { customDate } = useContext(DateContext)!
  const [title, setTitle] = useState<string>('')
  const [description, setDescription] = useState<string>('')
  const [date, setDate] = useState<Date>(customDate)
  const [url, setUrl] = useState<string>('')
  const [street, setStreet] = useState<string>('')
  const [city, setCity] = useState<string>('')
  const [isOnlineEvent, setIsOnlineEvent] = useState<boolean>(false)
  const [hasMaxCapacity, setHasMaxCapacity] = useState<boolean>(true)
  const [maxCapacity, setMaxCapacity] = useState<number>(100)

  const emptyFields = () => {
    const noLocationFields = () => {
      return !url.length && (!street.length || !city.length)
    }

    return !title.length || !description.length || noLocationFields() || !date
  }

  const handleCreateMeetup = (e: React.SyntheticEvent) => {
    e.preventDefault()

    const meetup: Meetup = {
      id: uid(),
      title,
      description,
      date,
      online: isOnlineEvent,
      capacity: hasMaxCapacity ? maxCapacity : null,
      attending: 0,
      arranger: user,
      reviews: [],
      comments: [],
    }

    // if (!meetup.date) {
    //   meetup.date = customDate
    // }

    if (isOnlineEvent) {
      meetup.url = url
    } else {
      meetup.location = {
        street,
        city,
      }
    }

    setMeetups([...meetups, meetup])
    addMeetup(meetup)
    navigate(`/profile/${user.id}`)
  }

  const handleCancel = () => {
    navigate(`/profile/${user.id}`)
  }

  return (
    <main className={styles.createMeetup}>
      <h1>CREATE A NEW MEETUP</h1>
      <form onSubmit={handleCreateMeetup} aria-label="create-form">
        <div className={styles.inputRow}>
          <label htmlFor="title">Title</label>
          <input
            value={title}
            onChange={(e) => setTitle(e.currentTarget.value)}
            maxLength={30}
            className={`${styles.textInput} ${styles.titleInput}`}
            id="title"
            type="text"
          />
        </div>
        <div className={styles.inputRow}>
          <label htmlFor="description">Description</label>
          <textarea
            value={description}
            onChange={(e) => setDescription(e.currentTarget.value)}
            maxLength={75}
            className={styles.textInput}
            id="description"
          />
        </div>
        <div className={styles.inputRow}>
          <label>Date</label>
          <DateTimePicker
            disableClock={true}
            minDate={customDate}
            onChange={setDate}
            value={date!}
            nativeInputAriaLabel="Date"
          />
        </div>
        <div className={styles.checkboxRow}>
          <label>
            <input
              name="online"
              type="checkbox"
              onChange={() => setIsOnlineEvent(!isOnlineEvent)}
              checked={isOnlineEvent}
            />
            Online event
          </label>
        </div>
        {isOnlineEvent ? (
          <div className={styles.inputRow}>
            <label htmlFor="url">
              Meetup url (will be shown to attendants when they sign up))
            </label>
            <input
              value={url}
              onChange={(e) => setUrl(e.currentTarget.value)}
              id="url"
              type="text"
            />
          </div>
        ) : (
          <div>
            <h4>Address</h4>
            <div className={styles.inputRow}>
              <label htmlFor="street">Street</label>
              <input
                value={street}
                onChange={(e) => setStreet(e.currentTarget.value)}
                id="street"
                type="text"
              />
            </div>
            <div className={styles.inputRow}>
              <label htmlFor="city">City</label>
              <input
                value={city}
                onChange={(e) => setCity(e.currentTarget.value)}
                id="city"
                type="text"
              />
            </div>
          </div>
        )}
        <div className={styles.checkboxRow}>
          <label>
            <input
              checked={hasMaxCapacity}
              onChange={() => setHasMaxCapacity(!hasMaxCapacity)}
              name="capacity-true"
              type="checkbox"
            />
            Max capacity
          </label>
        </div>
        {hasMaxCapacity && (
          <div className={styles.inputRow}>
            <label htmlFor="max-capacity">Maximum number of attendants</label>
            <input
              value={maxCapacity}
              onChange={(e) => setMaxCapacity(+e.currentTarget.value)}
              aria-label="capacity-input"
              id="max-capacity"
              type="number"
              pattern="[0-9]*"
              step={10}
              min={0}
            />
          </div>
        )}
        <div className={styles.buttons}>
          <button onClick={handleCancel} type="button">
            Cancel
          </button>
          <button disabled={emptyFields()}>Create</button>
        </div>
      </form>
    </main>
  )
}

export default CreateMeetup
