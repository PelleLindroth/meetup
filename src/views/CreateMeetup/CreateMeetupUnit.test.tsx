import { getUserById } from '../../db'
import { renderWithRouter, mountWithPath } from '../../utils/testing-utils'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import CreateMeetup from '.'

describe('Create Meetup unit tests', () => {
  const user = getUserById('1')

  it('renders Create Meetup page correctly (smoke test)', () => {
    jest.useFakeTimers('modern')
    jest.setSystemTime(new Date(2021, 11, 24))

    const wrapper = mountWithPath(
      <CreateMeetup setMeetups={jest.fn()} user={user!} />,
      '/create',
      '/create'
    )

    expect(wrapper.find(CreateMeetup)).toMatchSnapshot()

    jest.useRealTimers()
  })
  it('renders a form with inputs and a submit button', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const createForm = screen.getByRole('form', { name: /create/ })

    expect(createForm).toBeInTheDocument()
  })
  it('renders empty inputs for title and description', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const titleInput = screen.getByRole('textbox', { name: /title/i })
    const descriptionInput = screen.getByRole('textbox', {
      name: /description/i,
    })

    expect(titleInput).toBeInTheDocument()
    expect(descriptionInput).toBeInTheDocument()
  })
  it('renders Datetimepicker with Christmas Eve 2021 as default', () => {
    jest.useFakeTimers('modern')
    jest.setSystemTime(new Date(2021, 11, 24))

    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const date = screen.getByLabelText('Date')

    expect(date).toHaveValue('2021-12-24T00:00')

    jest.useRealTimers()
  })
  it('renders unchecked checkbox for online event and empty text inputs for street and city', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const onlineCheckbox = screen.getByRole('checkbox', { name: /online/i })

    expect(onlineCheckbox).toBeInTheDocument()
    expect(onlineCheckbox).not.toBeChecked()

    const streetInput = screen.getByRole('textbox', { name: /street/i })
    const cityInput = screen.getByRole('textbox', { name: /city/i })

    expect(streetInput).toHaveTextContent('')
    expect(cityInput).toHaveTextContent('')
  })
  it('renders checked checkbox for max capacity and number input for max attendants with 100 as default', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const maxCapacityCheckbox = screen.getByRole('checkbox', {
      name: /max capacity/i,
    })
    const maxCapacityInput = screen.getByRole('spinbutton', {
      name: /capacity-input/,
    })

    expect(maxCapacityCheckbox).toBeChecked()
    expect(maxCapacityInput).toHaveValue(100)
  })
  it('renders an enabled cancel button and a disabled create button', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const cancelButton = screen.getByRole('button', { name: /cancel/i })
    const createButton = screen.getByRole('button', { name: /create/i })

    expect(cancelButton).toBeEnabled()
    expect(createButton).toBeDisabled()
  })
  it('shows url text input and hides address inputs if "online event" is checked', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const onlineCheckbox = screen.getByRole('checkbox', { name: /online/i })

    const streetInput = screen.getByRole('textbox', { name: /street/i })
    const cityInput = screen.getByRole('textbox', { name: /city/i })

    userEvent.click(onlineCheckbox)

    const urlInput = screen.getByRole('textbox', { name: /url/i })

    expect(urlInput).toBeInTheDocument()
    expect(streetInput).not.toBeInTheDocument()
    expect(cityInput).not.toBeInTheDocument()
  })
  it('hides number input for max capacity if "max capacity" is unchecked', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const maxCapacityCheckbox = screen.getByRole('checkbox', {
      name: /max capacity/i,
    })
    const maxCapacityInput = screen.getByRole('spinbutton', {
      name: /capacity-input/,
    })

    userEvent.click(maxCapacityCheckbox)

    expect(maxCapacityInput).not.toBeInTheDocument()
  })
  it('enables create button if no inputs on screen are empty', () => {
    renderWithRouter(<CreateMeetup setMeetups={jest.fn()} user={user!} />)

    const textInputs = screen.getAllByRole('textbox')

    textInputs.forEach((input: any) => {
      userEvent.type(input, 'hey')
    })

    const createButton = screen.getByRole('button', { name: /create/i })

    expect(createButton).toBeEnabled()
  })
})
