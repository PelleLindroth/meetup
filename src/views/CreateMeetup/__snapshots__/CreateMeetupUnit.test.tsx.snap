// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Meetup unit tests renders Create Meetup page correctly (smoke test) 1`] = `
<CreateMeetup
  meetups={Array []}
  setMeetups={[MockFunction]}
  user={
    Object {
      "attending": Array [
        "1",
        "2",
      ],
      "email": "tjalle@yahoo.com",
      "firstName": "Tjalle",
      "id": "1",
      "lastName": "Bergkvist",
      "password": "grillkorv",
      "reviewed": Array [
        "1",
      ],
    }
  }
>
  <main
    className="createMeetup"
  >
    <h1>
      CREATE A NEW MEETUP
    </h1>
    <form
      aria-label="create-form"
      onSubmit={[Function]}
    >
      <div
        className="inputRow"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <input
          className="textInput titleInput"
          id="title"
          maxLength={30}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="inputRow"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="textInput"
          id="description"
          maxLength={75}
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="inputRow"
      >
        <label>
          Date
        </label>
        <DateTimePicker
          calendarIcon={
            <svg
              className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="15"
                width="15"
                x="2"
                y="2"
              />
              <line
                x1="6"
                x2="6"
                y1="0"
                y2="4"
              />
              <line
                x1="13"
                x2="13"
                y1="0"
                y2="4"
              />
            </svg>
          }
          clearIcon={
            <svg
              className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="4"
                x2="15"
                y1="4"
                y2="15"
              />
              <line
                x1="15"
                x2="4"
                y1="4"
                y2="15"
              />
            </svg>
          }
          closeWidgets={true}
          disableClock={true}
          isCalendarOpen={null}
          isClockOpen={null}
          maxDetail="minute"
          minDate={2021-12-23T23:00:00.000Z}
          nativeInputAriaLabel="Date"
          onChange={[Function]}
          openWidgetsOnFocus={true}
          value={2021-12-23T23:00:00.000Z}
        >
          <div
            className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled"
            onFocus={[Function]}
          >
            <div
              className="react-datetime-picker__wrapper"
            >
              <DateTimeInput
                className="react-datetime-picker__inputGroup"
                isWidgetOpen={null}
                maxDetail="minute"
                minDate={2021-12-23T23:00:00.000Z}
                name="datetime"
                nativeInputAriaLabel="Date"
                onChange={[Function]}
                value={2021-12-23T23:00:00.000Z}
              >
                <div
                  className="react-datetime-picker__inputGroup"
                  onClick={[Function]}
                >
                  <NativeInput
                    ariaLabel="Date"
                    key="time"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    name="datetime"
                    onChange={[Function]}
                    value={2021-12-23T23:00:00.000Z}
                    valueType="minute"
                  >
                    <input
                      aria-label="Date"
                      max="275760-09-13T02:00"
                      min="2021-12-24T00:00"
                      name="datetime"
                      onChange={[Function]}
                      onFocus={[Function]}
                      step={60}
                      style={
                        Object {
                          "position": "absolute",
                          "visibility": "hidden",
                          "zIndex": "-999",
                        }
                      }
                      type="datetime-local"
                      value="2021-12-24T00:00"
                    />
                  </NativeInput>
                  <MonthInput
                    className="react-datetime-picker__inputGroup"
                    inputRef={
                      Object {
                        "current": <input
                          autocomplete="off"
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                          data-input="true"
                          inputmode="numeric"
                          max="12"
                          min="12"
                          name="month"
                          placeholder="--"
                          type="number"
                          value="12"
                        />,
                      }
                    }
                    key="month"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    required={null}
                    value="12"
                    year="2021"
                  >
                    <Input
                      className="react-datetime-picker__inputGroup"
                      inputRef={
                        Object {
                          "current": <input
                            autocomplete="off"
                            class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                            data-input="true"
                            inputmode="numeric"
                            max="12"
                            min="12"
                            name="month"
                            placeholder="--"
                            type="number"
                            value="12"
                          />,
                        }
                      }
                      max={12}
                      min={12}
                      name="month"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={null}
                      value="12"
                    >
                      <input
                        autoComplete="off"
                        className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                        data-input="true"
                        inputMode="numeric"
                        key="input"
                        max={12}
                        min={12}
                        name="month"
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        onKeyUp={[Function]}
                        placeholder="--"
                        required={null}
                        type="number"
                        value="12"
                      />
                    </Input>
                  </MonthInput>
                  <Divider
                    key="separator_1"
                  >
                    <span
                      className="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                  </Divider>
                  <DayInput
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup"
                    inputRef={
                      Object {
                        "current": <input
                          autocomplete="off"
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                          data-input="true"
                          inputmode="numeric"
                          max="31"
                          min="24"
                          name="day"
                          placeholder="--"
                          type="number"
                          value="24"
                        />,
                      }
                    }
                    key="day"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    month="12"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    required={null}
                    value="24"
                    year="2021"
                  >
                    <Input
                      autoFocus={false}
                      className="react-datetime-picker__inputGroup"
                      inputRef={
                        Object {
                          "current": <input
                            autocomplete="off"
                            class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                            data-input="true"
                            inputmode="numeric"
                            max="31"
                            min="24"
                            name="day"
                            placeholder="--"
                            type="number"
                            value="24"
                          />,
                        }
                      }
                      max={31}
                      min={24}
                      name="day"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={null}
                      value="24"
                    >
                      <input
                        autoComplete="off"
                        autoFocus={false}
                        className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                        data-input="true"
                        inputMode="numeric"
                        key="input"
                        max={31}
                        min={24}
                        name="day"
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        onKeyUp={[Function]}
                        placeholder="--"
                        required={null}
                        type="number"
                        value="24"
                      />
                    </Input>
                  </DayInput>
                  <Divider
                    key="separator_2"
                  >
                    <span
                      className="react-datetime-picker__inputGroup__divider"
                    >
                      /
                    </span>
                  </Divider>
                  <YearInput
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup"
                    inputRef={
                      Object {
                        "current": <input
                          autocomplete="off"
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                          data-input="true"
                          inputmode="numeric"
                          max="275760"
                          min="2021"
                          name="year"
                          placeholder="----"
                          step="1"
                          type="number"
                          value="2021"
                        />,
                      }
                    }
                    key="year"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    required={null}
                    value="2021"
                    valueType="day"
                  >
                    <Input
                      autoFocus={false}
                      className="react-datetime-picker__inputGroup"
                      inputRef={
                        Object {
                          "current": <input
                            autocomplete="off"
                            class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                            data-input="true"
                            inputmode="numeric"
                            max="275760"
                            min="2021"
                            name="year"
                            placeholder="----"
                            step="1"
                            type="number"
                            value="2021"
                          />,
                        }
                      }
                      max={275760}
                      min={2021}
                      name="year"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      placeholder="----"
                      required={null}
                      step={1}
                      value="2021"
                    >
                      <input
                        autoComplete="off"
                        autoFocus={false}
                        className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                        data-input="true"
                        inputMode="numeric"
                        key="input"
                        max={275760}
                        min={2021}
                        name="year"
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        onKeyUp={[Function]}
                        placeholder="----"
                        required={null}
                        step={1}
                        type="number"
                        value="2021"
                      />
                    </Input>
                  </YearInput>
                  <Divider
                    key="separator_3"
                  >
                    <span
                      className="react-datetime-picker__inputGroup__divider"
                    >
                       
                    </span>
                  </Divider>
                  <Hour12Input
                    amPm="am"
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup"
                    inputRef={
                      Object {
                        "current": <input
                          autocomplete="off"
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                          data-input="true"
                          inputmode="numeric"
                          max="12"
                          min="1"
                          name="hour12"
                          placeholder="--"
                          type="number"
                          value="12"
                        />,
                      }
                    }
                    key="hour12"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    required={null}
                    showLeadingZeros={false}
                    value="0"
                  >
                    <Input
                      autoFocus={false}
                      className="react-datetime-picker__inputGroup"
                      inputRef={
                        Object {
                          "current": <input
                            autocomplete="off"
                            class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                            data-input="true"
                            inputmode="numeric"
                            max="12"
                            min="1"
                            name="hour12"
                            placeholder="--"
                            type="number"
                            value="12"
                          />,
                        }
                      }
                      max={12}
                      maxDate={+275760-09-13T00:00:00.000Z}
                      min={1}
                      minDate={2021-12-23T23:00:00.000Z}
                      name="hour12"
                      nameForClass="hour"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={null}
                      showLeadingZeros={false}
                      value="12"
                    >
                      <input
                        autoComplete="off"
                        autoFocus={false}
                        className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
                        data-input="true"
                        inputMode="numeric"
                        key="input"
                        max={12}
                        min={1}
                        name="hour12"
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        onKeyUp={[Function]}
                        placeholder="--"
                        required={null}
                        type="number"
                        value="12"
                      />
                    </Input>
                  </Hour12Input>
                  <Divider
                    key="separator_4"
                  >
                    <span
                      className="react-datetime-picker__inputGroup__divider"
                    >
                      :
                    </span>
                  </Divider>
                  <MinuteInput
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup"
                    hour="0"
                    inputRef={
                      Object {
                        "current": <input
                          autocomplete="off"
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                          data-input="true"
                          inputmode="numeric"
                          max="59"
                          min="0"
                          name="minute"
                          placeholder="--"
                          type="number"
                          value="0"
                        />,
                      }
                    }
                    key="minute"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    required={null}
                    showLeadingZeros={true}
                    value="0"
                  >
                    <Input
                      autoFocus={false}
                      className="react-datetime-picker__inputGroup"
                      inputRef={
                        Object {
                          "current": <input
                            autocomplete="off"
                            class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                            data-input="true"
                            inputmode="numeric"
                            max="59"
                            min="0"
                            name="minute"
                            placeholder="--"
                            type="number"
                            value="0"
                          />,
                        }
                      }
                      max={59}
                      maxDate={+275760-09-13T00:00:00.000Z}
                      min={0}
                      minDate={2021-12-23T23:00:00.000Z}
                      name="minute"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={null}
                      showLeadingZeros={true}
                      value="0"
                    >
                      <span
                        className="react-datetime-picker__inputGroup__leadingZero"
                        key="leadingZero"
                      >
                        0
                      </span>
                      <input
                        autoComplete="off"
                        autoFocus={false}
                        className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute react-datetime-picker__inputGroup__input--hasLeadingZero"
                        data-input="true"
                        inputMode="numeric"
                        key="input"
                        max={59}
                        min={0}
                        name="minute"
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyPress={[Function]}
                        onKeyUp={[Function]}
                        placeholder="--"
                        required={null}
                        type="number"
                        value="0"
                      />
                    </Input>
                  </MinuteInput>
                  <Divider
                    key="separator_5"
                  >
                    <span
                      className="react-datetime-picker__inputGroup__divider"
                    >
                       
                    </span>
                  </Divider>
                  <AmPm
                    autoFocus={false}
                    className="react-datetime-picker__inputGroup"
                    inputRef={
                      Object {
                        "current": <select
                          class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                          data-input="true"
                          data-select="true"
                          name="amPm"
                        >
                          <option
                            value="am"
                          >
                            AM
                          </option>
                          <option
                            value="pm"
                          >
                            PM
                          </option>
                        </select>,
                      }
                    }
                    key="ampm"
                    maxDate={+275760-09-13T00:00:00.000Z}
                    minDate={2021-12-23T23:00:00.000Z}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    placeholder="--"
                    required={null}
                    value="am"
                  >
                    <select
                      className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
                      data-input="true"
                      data-select="true"
                      name="amPm"
                      onChange={[Function]}
                      required={null}
                      value="am"
                    >
                      <option
                        value="am"
                      >
                        AM
                      </option>
                      <option
                        value="pm"
                      >
                        PM
                      </option>
                    </select>
                  </AmPm>
                </div>
              </DateTimeInput>
              <button
                className="react-datetime-picker__clear-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                className="react-datetime-picker__calendar-button react-datetime-picker__button"
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                <svg
                  className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height={19}
                  stroke="black"
                  strokeWidth={2}
                  viewBox="0 0 19 19"
                  width={19}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </DateTimePicker>
      </div>
      <div
        className="checkboxRow"
      >
        <label>
          <input
            checked={false}
            name="online"
            onChange={[Function]}
            type="checkbox"
          />
          Online event
        </label>
      </div>
      <div>
        <h4>
          Address
        </h4>
        <div
          className="inputRow"
        >
          <label
            htmlFor="street"
          >
            Street
          </label>
          <input
            id="street"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="inputRow"
        >
          <label
            htmlFor="city"
          >
            City
          </label>
          <input
            id="city"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="checkboxRow"
      >
        <label>
          <input
            checked={true}
            name="capacity-true"
            onChange={[Function]}
            type="checkbox"
          />
          Max capacity
        </label>
      </div>
      <div
        className="inputRow"
      >
        <label
          htmlFor="max-capacity"
        >
          Maximum number of attendants
        </label>
        <input
          aria-label="capacity-input"
          id="max-capacity"
          min={0}
          onChange={[Function]}
          pattern="[0-9]*"
          step={10}
          type="number"
          value={100}
        />
      </div>
      <div
        className="buttons"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          disabled={true}
        >
          Create
        </button>
      </div>
    </form>
  </main>
</CreateMeetup>
`;
